tests/Application.IntegrationTests/PooledContext.cs

```diff
@@ -22,9 +22,20 @@ public void Dispose()
         Contexts.Add(_context);
     }
 
-    public TResult Execute<TResult>(Func<Context, TResult> func)
-        where TResult : Z3Object
+    public Solver CreateSolver(Func<Context, Solver> func)
     {
-        return _context.Execute(func);
+        return _context.CreateSolver(func);
+    }
+
+    public TSort CreateSort<TSort>(Func<Context, TSort> func)
+        where TSort : Sort
+    {
+        return _context.CreateSort(func);
+    }
+
+    public TExpr CreateExpr<TExpr>(Func<Context, TExpr> func)
+        where TExpr : Expr
+    {
+        return _context.CreateExpr(func);
```

Comment:
It's pretty clear to me now that I should not have killed `IContextHandle` from before, and should instead allow that to vary separately from the new `IContext`, which can just have one by composition. I thought I could do everything on one interface but it is actually stupid because I will have to wrap all the methods like this, and it also means I would have to unnecessarily make things public that otherwise don't have to be on my following work. If everything else is ok on this PR I will do another small follow-up with that change.

Reply:
Never mind, just done it on this PR otherwise other things don't really make much sense.